src/Persistence/IteratorQuery.php

```diff
@@ -0,0 +1,393 @@
+<?php
+
+declare(strict_types=1);
+
+namespace atk4\data\Persistence;
+
+use atk4\data\Exception;
+use atk4\data\Field;
+use atk4\data\Model;
+use atk4\data\Model\Scope\Condition;
+use atk4\data\Persistence;
+
+/**
+ * Class to perform queries on Array_ persistence.
+ */
+class IteratorQuery extends AbstractQuery
```

Comment:
ArrayIteratorQuery?

Reply:
No because it is used for various iterators for persistences.
For `Persistence\Array_` we pass `\ArrayIterator`
For `Persistence\Csv` we pass `\SplFileObject` which is also an iterator